First Commit

NByamane opened the pull request:
基本的なアプリ挙動部分の作成を行いました。
以下対応していない点です。

- SP対応
- フォーム入力部分のバリデーション
- その他CSSで整えるべき見栄え

NByamane committed:
first_commit

基本的なアプリ挙動部分の作成を行いました。
以下対応していない点です。

・SP対応
・フォーム入力部分のバリデーション
・その他CSSで整えるべき見栄え

NBmaeda commented:
対応ありがとうございます！
とりあえず、ざっくり見た上でのコメントをします💡

- `node_modules`がgitでバージョン管理の対象になっています。Viteのスターターファイルに含まれていないようなら、`.gitignore`を自分で作って指定した方が良いです。どのファイルをバージョン管理の対象にするか、というのは気をつけてあげてください。
- 残りの対応ですが、SP対応とCSS調整は良いのでバリデーションのみ実装をお願いします。
- `ShortenLinkInputErea`ですが、純粋に「Area」を誤字ってそうです🥹

概ね良い感じだと思います！🎉
引き続きよろしくお願いします🙇

NByamane commented:
@NBmaeda 
お疲れ様です🙇‍♀️
レビューありがとうございました🎉
修正対応いたします🫡

いくつか質問させてください🙂
① `node_modules`がgitでの管理対象となっているということですが、本来のViteスターターファイルには`node_modules`は含まれているものなんでしょうか？（私のVite導入方法が間違ってたとかかなぁと🤔）
② `node_modules`は基本的にgitでバージョン管理しない方が良いということでしょうか？👀確かに今まで関わってきたプロジェクトのファイルではよく`.gitignore`ファイルを見かけていたのでそういった管理するしないを分けることは普通なのだろうとは思うのですが、node周りのバージョンだったりは共同開発の場合、全体で合わせるべきのように感じます🤔管理するしないはどういう考えの元使い分けるのか、基準などあればお伺いしたいです🙏（この場合、なぜ`node_modelues`を含めない方が良いのかがピンときていません😇ｽﾐﾏｾﾝ）

上記2点、お手隙の際にご確認いただけますと幸いです🙇‍♀️
よろしくお願いいたします☺️

NBmaeda commented:
@NByamane 

>  ① `node_modules`がgitでの管理対象となっているということですが、本来のViteスターターファイルには`node_modules`は含まれているものなんでしょうか？（私のVite導入方法が間違ってたとかかなぁと🤔）

理由は後述しますが、`node_modules`はどのスターターファイル・プロジェクトでも含まれていないと思います。
[公式のスターターガイド](https://ja.vitejs.dev/guide/)の手順で確認したら`.gitignore`は含まれていたので、もしかしたら誤って削除してしまったかもしれません。


> ② `node_modules`は基本的にgitでバージョン管理しない方が良いということでしょうか？👀

これは「バージョン管理しない方が良い」です。
いちばん大きな理由はそのサイズで`url-shortening-api-master`で言えば、全体が`139MB`に対して`node_modules`は`115MB`となっています(80%超🤯)。
パッケージが増えるほどファイル数もサイズも増大していきますし、常にバージョン管理を行うには重すぎるのです。

> node周りのバージョンだったりは共同開発の場合、全体で合わせるべき

これはまさにその通りです。
`node_modules`をバージョン管理しなくても全体で合わせることができるのは、`package.json`のお陰です。
`package.json`をバージョン管理しておくことで、`npm install`を行うだけで環境の再現ができますし、npm-scriptsで行う処理の統一なども行うことができます。

なのでまとめると、以下のようになります。
- `node_modules`はバージョン管理しない方が良いため、`.gitignore`に記載する。
- その代わりに`package.json`をバージョン管理・メンテナンスし、環境の再現ができるようにする。

`.gitignore`にはAPI Keyのようなクレデンシャルが記載されているファイルも必ず含める必要があるので、覚えておいてください。
そうしないとリポジトリに重要な情報がダダ漏れとなります🥳

NByamane committed:
レビュー修正

・ShortenLinkInputEreaのスペルミスを修正（Area）
・バリデーション機能追加
・.gitignoreファイルを追加して、node_modules/以下を追跡から除外

NByamane commented:
@NBmaeda 
お疲れ様です🙇‍♀️
ご返答ありがとうございました✨

修正いたしましたので、ご確認お願いいたします🙏

NBmaeda reviewed (changes requested):
`ShortenLinkInputErea`および`.gitignore`の件は確認できました！

バリデーションに関しては入力がない場合のエラーは出るのですが、URLでないテキストは許容されるのでAPIから400エラーが返ってくる状況です。
URLの形式かどうかのバリデーションを追加してみてください💡

NByamane committed:
レビュー修正

URL形式が正しいかどうかのバリデーションを追加

NByamane commented:
@NBmaeda 
ご確認ありがとうございます🙇‍♀️
URL形式が正しいかどうかのバリデーションも追加しました🙋‍♀️
お手隙の際にご確認お願いいたします🍬

NBmaeda reviewed (approved):
URLのバリデーション確認できました！
問題ないと思います、お疲れ様でした👏
